src/screens/Elections/index.js

```diff
@@ -22,4 +31,214 @@ const propTypes = {
+      data={[...distinctPositions]}
+      keyExtractor={(item)=>item}
+      renderItem={({item})=> (
+        <View key={item} style={styles.candidateContainer}>
+          <View style={styles.informationContainer}>
+            <Text style={styles.positionText}>{item} </Text>
+            <Text style={styles.electionText}>November 6, 2018</Text>
+          </View>
+          <ScrollView horizontal={true}>
+            {props.candidates.map(candidate => (
+              (item === candidate.electionIds.toString())
+                ?
+                <View key={candidate.id}>
+                  <Data candidateId={candidate.id} goToCandidateDetail={props.goToCandidateDetail} />
+                </View>
+                :
+                null
+            ))}
+          </ScrollView>
+        </View>
+      )}/>
+  </View>
+);
+
+ElectionsView.propTypes = {
+  goToCandidateDetail: PropTypes.func,
+  candidates: PropTypes.array,
+};
+
+const DataContainer = (props) => (
+  <View style={styles.container}>
+    <View key={props.data.id} style={styles.container}>
+      <View style={styles.newsCard}>
+        <View style={styles.pictureBody}>
+          <TouchableOpacity onPress={props.goToCandidateDetail(props.data.id)}> 
+            <Image style={styles.candidatePicture}
+              source={{ uri: props.data.image}} 
+            />
+            <FavoriteContainer isFavorite={props.data.isFavorite}/>
+          </TouchableOpacity>
+        </View> 
+        <View style={styles.contentContainer}>
+          <View style={styles.contentBody}>
+            <Text style={styles.matchCardText}>
+              <Text style={styles.nameText}>{props.data.name}{'\n\n'}</Text>  
+              {props.data.matchPercent != undefined
+                ?
+                <Text style={styles.matchCardPercentText}>{props.data.matchPercent}{'%'}</Text>
+                :
+                <Text>{'No'}</Text>
+              }{' match'}
+            </Text>
+          </View>
+        </View>   
+      </View> 
+    </View>
+  </View>
+);
+DataContainer.propTypes = {
+  data: PropTypes.object,
+  id: PropTypes.string,
+  goToCandidateDetail: PropTypes.func,
+  name: PropTypes.string,
+  electionIds: PropTypes.string,
+  image: PropTypes.string,
+  matchPercent: PropTypes.number,
+};
+
+const FavoriteContainer = props => (
+  <View style={styles.container}>
+    {props.isFavorite ?
+      <View style={styles.favoriteBody}>
+        <Icon
+          name={'star'}
+          iconStyle={styles.favorite}
+          size={15}
+        />
+        <Text style={styles.favoriteText}>{'Favorite'}</Text>
+      </View> 
+      :
+      null
+    }
+  </View>
+);
+FavoriteContainer.propTypes = {
+  isFavorite: PropTypes.bool,
+};
+
+const Data = connect(
+  (state, ownProps) => ({
+    data: getCandidateData(state, ownProps.candidateId),
+  }),
+)(DataContainer);
+
+const ScreenWithAuthentication = WithAuthentication('logout')(ElectionsView);
+
+const Container = compose(
```

Comment:
I feel like it can be hard sometimes to find the code you're looking for in a large file. Personally I prefer smaller files with one component each and in other places we've broken out the redux containers into their own file.

Reply:
I put my components into different files broken by redux containers and renaming it to viewselectors.

Inside ScreenContainer.js there is one redux and I'm not sure how to put it inside viewSelectors.js if necessary.

Reply:
`export const Data = connect(
  (state, ownProps) => ({
    data: getCandidateData(state, ownProps.candidateId),
  }),
)(Candidate);
`